coordinates/Coordinates/DirectionCoordinate.cc

```diff
@@ -1619,88 +1619,81 @@ Bool DirectionCoordinate::toMix2(Vector<Double>& out,
-   Vector<String> unitsCanon(worldAxisUnits());
-//
-   Vector<String> namesOut(worldAxisNames().copy());
-   Vector<String> unitsOut(worldAxisUnits().copy());
-   fourierUnits (namesOut[0], unitsOut[0], unitsCanon[0], Coordinate::DIRECTION, 0,
-                 units[0], names[0]);
-   fourierUnits (namesOut[1], unitsOut[1], unitsCanon[1], Coordinate::DIRECTION, 1,
-                 units[1], names[1]);
-
-// Make a copy of ourselves and set the new units
+Coordinate* DirectionCoordinate::makeFourierCoordinate(
+    const Vector<Bool>& axes, const Vector<Int>& shape
+) const {
+    // axes says which axes in the coordinate are to be transformed
+    // shape is the shape of the image for all axes in this coordinate
+    AlwaysAssert(nPixelAxes()==2, AipsError);
+    AlwaysAssert(nWorldAxes()==2, AipsError);
+
+    if (axes.nelements() != 2) {
+        set_error ("Invalid number of specified axes");
+        return 0;
+    }
+    if (!axes[0] || !axes[1]) {
+        set_error ("You must specify both axes of the DirectionCoordinate to transform");
+        return 0;
+    }
+    if (shape.nelements() != 2) {
+        set_error ("Invalid number of elements in shape");
+        return 0;
+    }
 
-   DirectionCoordinate dc = *this;
-   if (!dc.setWorldAxisUnits(unitsCanon)) {
-      set_error ("Could not set world axis units");
-      return 0;
-   }
+    // Find names and units for Fourier coordinate and units to set 
+    // for this DirectionCoordinate 
+    Vector<String> names = worldAxisNames().copy();
+    Vector<String> units = worldAxisUnits().copy();
+    // Not copying causes a reference to this object's units to be passed
+    // in to the fourierUnits() call below and they can be changed.
+    // That would be bad.
+    Vector<String> unitsCanon = worldAxisUnits().copy();
+
+    Vector<String> namesOut(worldAxisNames().copy());
+    Vector<String> unitsOut(worldAxisUnits().copy());
+    fourierUnits(
+        namesOut[0], unitsOut[0], unitsCanon[0], Coordinate::DIRECTION, 0,
+        units[0], names[0]
+    );
+    fourierUnits(
+        namesOut[1], unitsOut[1], unitsCanon[1], Coordinate::DIRECTION, 1,
+        units[1], names[1]
+    );
+    // Make a copy of ourselves and set the new units
+    DirectionCoordinate dc = *this;
+    if (!dc.setWorldAxisUnits(unitsCanon)) {
+        set_error ("Could not set world axis units");
+        return 0;
+    }
+    // Create a LinearXform to do the inversion
+    Vector<Double> cdelt = dc.increment().copy();
+    dc.fromCurrent(cdelt);
+    LinearXform linear(dc.referencePixel(), cdelt, dc.linearTransform());
 
-// Create a LinearXform to do the inversion
+    // Now make the new output LinearCoordinate.  
 
-   Vector<Double> cdelt(dc.increment().copy());
-   fromCurrent(cdelt);
-   LinearXform linear(dc.referencePixel(), cdelt, dc.linearTransform());
+    Vector<Double> crpix(2), scale(2), crval(2,0.0);
+    crpix[0] = Int(shape[0] / 2);
+    crpix[1] = Int(shape[1] / 2);
```

Comment:
I didn't look at this in detail. The only change to these lines was the indentation (lines 1683 and 1684 were the line numbers in the original file) as part of my reformatting commit. Off the top of my head, it would seem OK to remove the explicit Int cast since shape is a Vector<Int> and, I assume the compiler interprets 2 as an Int. However, these lines were not responsible for the bug which motivated this fix, so in the interest of not changing code that appears to work (other than reformatting it), I'd prefer to leave this as is for now even though I agree, it is probably not written as compactly as it could be. But indeed, it was a good find.